userdocs/diagnostics/multiple-class-inheritance.md

```diff
@@ -0,0 +1,37 @@
+# Multiple Inheritance of Classes
+
+In some programming languages, a class can have multiple superclasses. This feature is called multiple inheritance and is unsupported in Swift. However, using protocols, one can achieve a similar effect as multiple inheritance.
+
+Consider the following example with a class `Bird` conforming to two protocols `Animal` and `Flyable`.
```

Comment:
```suggestion
Consider the following example:
```

While this example is simple, I don't think that it's the most effective demonstration here of the concept of protocol composition. Here's why:

The classic `Animal` example is often used because there is a well-known taxonomy among species that doesn't require explanation, because the reader can be expected to know this from grade school. However, that taxonomy precisely doesn't allow the kind of composition for which you might want to use multiple inheritance: no animal is both an amphibian and a mammal, for example.

Moreover, using an `Animal` example could be misleading to the reader whose mind goes to taxonomy, since inheriting from both `Animal` and `Mammal` notably would _not_ be an example of multiple inheritance, because `Mammal` would inherit from `Animal`.

Finally, nothing in the text here demonstrates a rationale for why `Flyable` exists; I don't think this evokes any ideas in the reader like `Animal` does, so to justify the protocol you'd have to provide some rationale why you'd want to model `Flyable`s in code orthogonally to the species taxonomy--which is certainly doable, but the example becomes much wordier.

I'm sure there are many commonly used examples from C++ pedagogy, say, that could be drawn upon for inspiration here. I don't know of any, but one example I can think of from the top of my head would be:

```swift
protocol Utensil { var name: String }

protocol ServingUtensil: Utensil { func serve() }
protocol Spoon: Utensil { func scoop() }
protocol Fork: Utensil { func spear() }

struct Ladle: ServingUtensil, Spoon { ... }
struct CarvingFork: ServingUtensil, Fork { ... }
struct Spork: Spoon, Fork { ... }
```

Reply:
A reasonable suggestion, I will adapt your suggestions by using the utensil example.